contrib/swagger.yaml

```diff
@@ -0,0 +1,439 @@
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+    post:
+      summary: Create a new code snippet.
+      description: |
+        The Snippets endpoint creates a new code snippet.
+      parameters:
+        - name: snippet
+          in: body
+          schema:
+            $ref: '#/definitions/Snippet'
+      responses:
+        201:
+          schema:
+            $ref: '#/definitions/Snippet'
+          description: Created. A new Snippet has been successfully created.
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. Returned when a user tries to create a non-anonymous
+            snippet without authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. Returned when an authorized user tries to create a snippet
+            on behalf of another user.
+
+  /snippets/{id}:
+    get:
+      summary: Retrieve information about the specific code snippet.
+      description: |
+        The Snippet endpoint returns information about the code snippet uniquely
+        identified by the given id value.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+      responses:
+        200:
+          description: A Snippet object
+          schema:
+            $ref: '#/definitions/Snippet'
+        400:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Bad request.
+        401:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Unauthorized. User tries to retrieve a private snippet without
+            authorizing first.
+        403:
+          schema:
+            $ref: '#/definitions/Error'
+          description: |
+            Forbidden. User tries to retrieve a private snippet by another author.
+        404:
+          schema:
+            $ref: '#/definitions/Error'
+          description: Snippet not found.
+    patch:
+      summary: Create a new changeset of the snippet.
+      description: Create a new changeset of the snippet.
+      parameters:
+        - name: id
+          type: integer
+          minimum: 1
+          format: int64
+          required: true
+          in: path
+          description: Unique identifier representing the specific code snippet.
+        - name: content
+          in: body
+          schema:
+            $ref: '#/definitions/Changeset'
```

Comment:
I was going to say only new ``changesets`` can be created, but it makes sense to support update for ``title`` and ``tags`` fields at least too.

At the same time, change of ``author_id`` must be prohibited. I'm not sure about change of ``syntax``: it's currently set on a ``snippet``, not ``changeset``, i.e. we should prohibit changes of both independently, otherwise it's possible for them to be not aligned.